data/datasets/TSSB-3M/generate_dataset.py

```diff
@@ -0,0 +1,133 @@
+"""Convert the source TSSB-3M  dataset to instruction data
+"""
+
+import json
+import re
+from os.path import join
+
+from tqdm import tqdm
+
+TEMPLATE = """User: Find the bug in the following code:
```

Comment:
maybe we should add random synonyms for the prompts to enrich the data

Reply:
> maybe we should add random synonyms for the prompts to enrich the data

Thank you for your suggestion. It has been modified to randomly generate synonymous prompts(instructions) from a  list